src/GovUk.Education.ExploreEducationStatistics.Content.Api/appsettings.json

```diff
@@ -5,5 +5,20 @@
       "Microsoft": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "MemoryCache": {
+    "Enabled": true,
+    "MaxCacheSizeMb": "50",
+    "ExpirationScanFrequencySeconds": "360",
+    "Configurations": {
+      "GetRelease": {
+        "DurationInSeconds": 30,
+        "ExpirySchedule": "*/30 * * * *"
+      },
+      "GetLatestRelease": {
+        "DurationInSeconds": 15,
+        "ExpirySchedule": "*/30 * * * *"
+      }
+    }
```

Comment:
Can't say I'm a big fan of this approach. It's too granular and means that we're having to add repetitive bits of config all across the code and ARM template for each additional key. Think this would make introducing new caches a bit of a pain and could be error prone. It also sets a convention that would be inevitably repeated for other cache types too (i.e. blob caching) and potentially become a viral anti-pattern.

Additionally, I'm not a big fan of the configuration being apart from the attribute itself. We're making people look in more places than is necessary and could cause bugs. Ideally, the configuration should just be as close as possible to where it's used.

As I see it, all we want to do is customise caching duration for tests so having such granular configuration is just overkill. From what we're doing in the tests, the test duration is set globally in `.env` and I can't think of any real reason why we'd want to have different durations for specific cases as all of this is being artificially set by the test environment.

Would suggest two approaches:

### Use overrides instead of per-key configuration

Would suggest the following that we get rid of key-specific `Configurations` and replace it with an optional `Overrides` property to override things globally during tests e.g:

  ```json
  "Overrides": {
    "DurationInSeconds": 2
  }
  ```

We'd just set properties as normal on `MemoryCache` attributes and if any of the global overrides are set, these take precedent instead.

### Disable memory caching for tests

It doesn't seem like we're even testing that caching works as part of the UI tests, so why go to the trouble of having more configuration than we need?

Unless we're planning on writing some tests to actually exercise this functionality, feels like it'd just be easier to get rid of all this configuration and set `MemoryCache.Enabled` to false for test environments.

Reply:
Yup I see your point here.  I've gone with the "Overrides" approach rather than disabling the MemoryCache altogether for particular environments - if there was some show-stopping problem with it, I wouldn't want it to sail by local and Dev environments because it's disabled there.